internal/core/services/room_manager/status.go

```diff
@@ -0,0 +1,60 @@
+package room_manager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+)
+
+// validStatusTransitions this map has all possible status changes for a game
+// room.
+var validStatusTransitions = map[game_room.GameRoomStatus]map[game_room.GameRoomStatus]struct{}{
+	game_room.GameStatusPending: {
+		game_room.GameStatusReady:   struct{}{},
+		game_room.GameStatusUnready: struct{}{},
+		game_room.GameStatusError:   struct{}{},
+	},
```

Comment:
When `DeleteRoom` is called in RoomManager it sets the status as terminating. Could we not get a case where it tries to set the status from pending to terminating?

Reply:
I think the best here is to permit any state to transition to terminating since we could have some "force-delete" scenarios. I'll update it.